### PR TITLE
Imagito shleifer

### PR DESCRIPTION
**Pull request by sshleifer**
We dont need 'params.txt'.
This works for me on GCP.
A few notes:
- K80 has no FP16 support so we need to reattach GPUs at some point.
- run `pip install funcy ipdb dill`

- Failed to get on Stanford slurm cluster, maybe only works on campus idk, will check.

### CODE REVIEW

File: fastai/imagito/train_imagito.py [hunk 1]
```diff
@@ -106,29 +107,29 @@ def main(
     lr *= bs_rat
 
     m = globals()[arch]
+    # save params to file like experiments/2019-05-12_22:10/params.pkl
+    now = get_date_str(seconds=False)
+    Path('experiments').mkdir(exist_ok=True)
+    model_dir = Path(f'experiments/{now}')
```
Comment (ecprokop) [inline fastai/imagito/train_imagito.py:107]: +1, was going to do this as well

File: fastai/imagito/utils.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import pandas as pd
+import re
+import time
+import pickle
+import gzip
+
+
+def get_date_str(seconds=False):
+    if seconds:
+        return time.strftime('%Y-%m-%d-%H%M:%S')
```
Comment (ecprokop) [inline fastai/imagito/utils.py]: I feel slightly better always having seconds just to guarantee we never have collisions

**Review by ecprokop — APPROVED** [review]
one minor comment to take or leave but LGTM

**Comment by ecprokop** [comment]
Fine with not having params.txt but what's the suggested alternative approach ?

Edit: looks like it was just pushed, now I see ParameterGrid

**Comment by sshleifer** [comment]
look at updated run_many.py ( http://run_many.py/ ).

It keeps the whole thing in python (so we dont go to text file and back).

I will add seconds!

617.966.4327

Leave Me Anonymous Feedback ( http://goo.gl/forms/ZTlmFBBTHq )

Sent via Superhuman ( https://sprh.mn/?vip=sshleifer@gmail.com )

**Pull request merged**